src/service/oneshot.rs

```diff
@@ -20,49 +18,43 @@ where
 // is ready, and then calling `Service::call` with the request, and
 // waiting for that `Future`.
 #[allow(missing_debug_implementations)]
+#[pin_project]
 pub struct Oneshot<S: Service<Req>, Req> {
+    #[pin]
     state: State<S, Req>,
 }
 
+#[pin_project(Replace, project = StateProj, project_replace = StateProjOwn)]
```

Comment:
It is not ideal that we need to have both `Replace` and `project_replace = <name>` as `project_replace = <name>` implies that a `project_replace` method is generated. 
I'll fix it soon.

Reply:
Do you mean I should wait to merge, or just some time later?

Reply:
Oh sorry, [a fix for this was already merged](https://github.com/taiki-e/pin-project/pull/243), but I forgot to release a new version.

Reply:
Updated to use a new version pin-project which includes a fix for this.